src/main/java/com/powsybl/openloadflow/equations/EquationSystem.java

```diff
@@ -118,7 +130,10 @@ public void onEquationTermChange(EquationTerm term, EquationTermEventType eventT
                 case EQUATION_TERM_ADDED:
                 case EQUATION_TERM_ACTIVATED:
                 case EQUATION_TERM_DEACTIVATED:
```

Comment:
couldn't you add a termsToRemove list and a termsToAdd list to avoid removing the whole equation and adding it again? well, maybe that's becoming too complex for the possible gain, it's sure easier like it is now - what do you think?